homework/index.js

```diff
@@ -22,21 +24,218 @@
+          ],
+        },
+        {
+          name: 'contributors',
+          tag: 'div',
+          attributes: {
+            class: 'contributors',
+          },
+          children: [
+            {
+              name: 'contributorsTitle',
+              tag: 'h3',
+              attributes: {
+                class: 'contributors__title',
+                text: 'contributors',
+              },
+            },
+            {
+              name: 'contributorsList',
+              tag: 'ul',
+              attributes: {
+                class: 'contributors__list',
+              },
+            },
+          ],
+        },
+      ],
+    },
+  ];
+
+  // Creates html elements from an object
+  function createStaticComponentsFromModel(model, root) {
+    model.forEach(parent => {
+      createAndAppend(parent.tag, root, parent.attributes);
+      const parentComponent = document.querySelector(`.${parent.attributes.class}`);
+
+      if (parent.children) {
+        createStaticComponentsFromModel(parent.children, parentComponent);
+      }
+    });
+  }
+
+  // Passes given data into dynamic html parts
+  function fillDataIntoComponents(selectedRepository) {
+    (function fillRepositoryDetailsTable() {
+      const tableModel = [
+        [
+          { class: 'label', text: 'Repository' },
+          { text: `<a href="${selectedRepository.html_url}">${selectedRepository.name}</a>` },
+        ],
+        [
+          { class: 'label', text: 'Description' },
+          {
+            text: selectedRepository.description
+              ? selectedRepository.description
+              : 'No description!',
+          },
+        ],
+        [
+          { class: 'label', text: 'Forks' },
+          { text: selectedRepository.forks ? selectedRepository.forks : 'No forks!' },
+        ],
+        [
+          { class: 'label', text: 'Updated At' },
+          { text: new Date(selectedRepository.updated_at).toLocaleString() },
+        ],
+      ];
+
+      const tableBody = document.querySelector('#table-body');
+
+      tableModel.forEach(row => {
+        const tableRow = createAndAppend('tr', tableBody);
+
+        row.forEach(column => createAndAppend('td', tableRow, column));
+      });
+    })();
+
+    (function fillContributorsList() {
+      fetchJSON(selectedRepository.contributors_url, (err, contributors) => {
+        const list = document.querySelector('.contributors__list');
+        if (err) {
+          createAndAppend('li', list, { class: 'text-error', text: err.message });
+        } else if (contributors.length) {
+          contributors.forEach(contributor => {
+            const listContent = `
+              <img class="contributors__avatar"
+                src="${contributor.avatar_url}" 
+                alt="${contributor.login}">
+              <span class="contributors__name">${contributor.login}</span>
+              <span class="contributors__badge">${contributor.contributions}</span>
+            `;
```

Comment:
So this is I guess why you needed to modify `createAndApppend`. But can't you use a function like this?

```js
  function createContributorModel(contributor) {
    return [
      {
        tag: 'li',
        attributes: { class: 'contributors__item' },
        children: [
          {
            tag: 'img',
            attributes: {
              class: 'contributors__avatar',
              src: contributor.avatar_url,
              alt: contributor.login,
            },
          },
          { tag: 'span', attributes: { class: 'contributors__name', text: contributor.login } },
          {
            tag: 'span',
            attributes: { class: 'contributors__badge', text: contributor.contributions },
          },
        ],
      },
    ];
  }
```

You could pass the return value to your `createStaticComponentsFromModel` function (to be renamed as it no longer deals with 'static' models). It would work best if it required a single element to be at the root level of the model, as React requires. In that way, you can return that root level element as the return value of the function, for further reference by the caller.